packages/legend-extension-mapping-generation/src/components/MappingGeneration_LegendStudioPlugin.tsx

```diff
@@ -0,0 +1,80 @@
+import packageJson from '../../package.json';
+import { MappingGenerationEditor } from './MappingGenerationEditor';
+import { MappingGenerationEditorState } from '../stores/MappingGenerationEditorState';
+import {
+  type EditorStore,
+  type DSLMapping_LegendStudioPlugin_Extension,
+  type ModelLoaderExtensionConfiguration,
+  LegendStudioPlugin,
+} from '@finos/legend-studio';
+import { assertErrorThrown, LogEvent } from '@finos/legend-shared';
+import type { Entity } from '@finos/legend-model-storage';
+
+const GENERATION_TYPE_NAME = `RELATIONAL_MAPPING_GENERATION`;
+const GENERATION_TYPE_FAILURE = `RELATIONAL_MAPPING_GENERATION_FAILURE`;
+
+export class MappingGeneration_LegendStudioPlugin
+  extends LegendStudioPlugin
+  implements DSLMapping_LegendStudioPlugin_Extension
+{
+  constructor() {
+    super(packageJson.extensions.studioPlugin, packageJson.version);
+  }
+
+  override getExtraModelLoaderExtensionConfigurations(): ModelLoaderExtensionConfiguration[] {
+    return [
+      {
+        key: GENERATION_TYPE_NAME,
+        load: async (editorStore: EditorStore): Promise<void> => {
```

Comment:
I would not load it in this matter. You should probably load this similar to the model generaiton `import` in schemaSetEditor. This loads the elements into the graph and then the user has the option of saving. 
The native loaders take a slight different approach which we will work on clearly a bit.